common/src/main/java/org/opensearch/ml/common/transport/mcpserver/requests/register/McpTool.java

```diff
@@ -0,0 +1,136 @@
+/*
+ *
+ *  * Copyright OpenSearch Contributors
+ *  * SPDX-License-Identifier: Apache-2.0
+ *
+ */
+
+package org.opensearch.ml.common.transport.mcpserver.requests.register;
+
+import static org.opensearch.core.xcontent.XContentParserUtils.ensureExpectedToken;
+
+import java.io.IOException;
+import java.util.Map;
+
+import org.opensearch.core.common.io.stream.StreamInput;
+import org.opensearch.core.common.io.stream.StreamOutput;
+import org.opensearch.core.common.io.stream.Writeable;
+import org.opensearch.core.xcontent.ToXContent;
+import org.opensearch.core.xcontent.ToXContentObject;
+import org.opensearch.core.xcontent.XContentBuilder;
+import org.opensearch.core.xcontent.XContentParser;
+
+import lombok.Data;
+import lombok.extern.log4j.Log4j2;
+
+/**
+ *  This class represents a tool that can be registered with OpenSearch. It contains information about the tool's name,
+ * description, parameters, and schema.
+ */
+@Log4j2
+@Data
+public class McpTool implements ToXContentObject, Writeable {
+    private static final String NAME_FIELD = "name";
+    private static final String DESCRIPTION_FIELD = "description";
+    private static final String PARAMS_FIELD = "params";
+    private static final String SCHEMA_FIELD = "schema";
+    private final String name;
```

Comment:
This is only for name or the tool type ? Current Agent Framework supports `name` and `type` when defining tool. That's for supporting multiple custom tools with same type. For example, user can define two MLModelTools with different name because they are for different purpose
```
   {
      "type": "MLModelTool",
     "name": "summarize"
    },
   {
      "type": "MLModelTool",
      "name": "translate"
    },
```

Reply:
Will change this in next PR.